emperor/support_files/js/view.js

```diff
@@ -337,6 +372,125 @@ DecompositionView.prototype._fastInit = function() {
   this.markers.push(cloud);
 };
 
+/**
+ * Parallel plots closely mirroring the shader enabled _fastInit calls
+ */
+DecompositionView.prototype._fastInitParallelPlot = function()
+{
+  var positions, colors, scales, opacities, visibilities, geometry, cloud;
+
+  // We're really just drawing a bunch of line strips...
+  // highly doubt shaders are necessary for this...
+  var vertexShader = `
+    attribute float scale;
+    attribute vec3 color;
+    attribute float opacity;
+    attribute float visible;
+    
+    varying vec3 vColor;
+    varying float vOpacity;
+    varying float vVisible;
+    
+    void main() {
+      vColor = color;
+      vOpacity = opacity;
+      vVisible = visible;
+    
+      gl_Position = projectionMatrix * modelViewMatrix * vec4(position, 1.0);
+      gl_PointSize = 2.0;
+    }
+    `;
+
+  var fragmentShader = `
+    precision mediump float;
+    varying vec3 vColor;
+    varying float vOpacity;
+    varying float vVisible;
+    
+    void main() {
+      if (vVisible <= 0.0)
```

Comment:
I think this should be  `(vVisible <= 0.0 || vOpacity <= 0)`.

Reply:
Updated